From c1d97639fc88d7d3e680473b79aab53b1315332e Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 20 Sep 2010 00:01:08 +0800
Subject: [PATCH 079/498] ENGR00131716 Camera: Not set mOverlay to zero when
 stop preview

Not set mOverlay to zero when stop preview due to ENGR00131682 commit

Signed-off-by: Sammy He <r62914@freescale.com>
---
 libcamera/CameraHal.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 94067d5..af1a841 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -910,7 +910,6 @@ void CameraHal::cameraPreviewStop()
 		usleep(30000);
             }
         }
-	mOverlay = 0;
 
 /* no need to DQBUF before STREAMOFF for UVC camera to improve performance */
 #if 0
@@ -939,8 +938,10 @@ void CameraHal::cameraPreviewStop()
 	cameraClose();
     }
 //    Mutex::Autolock lock(mLock);
-    mPreviewShowFrameThread.clear();
-    mPreviewCaptureFrameThread.clear();
+    if (mPreviewShowFrameThread != 0)
+        mPreviewShowFrameThread.clear();
+    if (mPreviewCaptureFrameThread != 0)
+        mPreviewCaptureFrameThread.clear();
 }
 
 bool CameraHal::previewEnabled()
-- 
1.8.0

